### PR TITLE
refactor: remove long

### PR DESCRIPTION
**Pull request by dundargoc**
long is 32-bits even on 64-bit windows which makes the type suboptimal
for a codebase meant to be cross-platform.

### CODE REVIEW

File: src/nvim/autocmd.c [hunk 1]
```diff
@@ -1694,7 +1694,7 @@ bool apply_autocmds_group(event_T event, char *fname, char *fname_io, bool force
   static int nesting = 0;
   AutoPat *ap;
   char *save_cmdarg;
-  long save_cmdbang;
+  int64_t save_cmdbang;
```
Comment (bfredl) [inline src/nvim/autocmd.c:1697]: `varnumber_T`

File: src/nvim/diff.c [hunk 2]
```diff
@@ -2466,8 +2466,8 @@ int diffopt_changed(void)
   int linematch_lines_new = 0;
   int diff_flags_new = 0;
   int diff_foldcolumn_new = 2;
-  long diff_algorithm_new = 0;
-  long diff_indent_heuristic = 0;
```
Comment (bfredl) [inline src/nvim/diff.c:2470]: these can be `int`

**Comment by bfredl** [comment]
in general `count` from ex commands seems to be capped at 999999999. so I think we can change `long`:s cargo cult copied from `count` ex argument (when these are not line numbers) to just `int` instead of `int64_t` (I assume the long comes from early vim ms dos builds where int could be limited to 16-bit).

**Comment by clason** [comment]
Needs a rebase -- @bfredl other comments?

File: src/nvim/drawline.c [hunk 3]
```diff
@@ -531,19 +531,19 @@ int win_signcol_width(win_T *wp)
 
 static inline void get_line_number_str(win_T *wp, linenr_T lnum, char *buf, size_t buf_len)
 {
-  long num;
-  char *fmt = "%*ld ";
+  linenr_T num;
+  char *fmt = "%*d ";
```
Comment (zeertzjq) [inline src/nvim/drawline.c:535]: `"%*" PRIdLINENR " "`

Comment (zeertzjq) [inline src/nvim/drawline.c:535]:
```suggestion
  char *fmt = "%*" PRIdLINENR " ";
```

File: src/nvim/drawline.c [hunk 4]
```diff
@@ -531,19 +531,19 @@ int win_signcol_width(win_T *wp)
...
+    num = lnum;
   } else {
     // 'relativenumber', don't use negative numbers
-    num = labs((long)get_cursor_rel_lnum(wp, lnum));
+    num = abs(get_cursor_rel_lnum(wp, lnum));
     if (num == 0 && wp->w_p_nu && wp->w_p_rnu) {
       // 'number' + 'relativenumber'
       num = lnum;
-      fmt = "%-*ld ";
+      fmt = "%-*d ";
```
Comment (zeertzjq) [inline src/nvim/drawline.c:546]:
```suggestion
      fmt = "%*" PRIdLINENR " ";
```

File: src/nvim/globals.h [hunk 5]
```diff
@@ -97,8 +97,8 @@ EXTERN struct nvim_stats_s {
 // up).
 #define DFLT_COLS       80              // default value for 'columns'
 #define DFLT_ROWS       24              // default value for 'lines'
-EXTERN int Rows INIT(= DFLT_ROWS);     // nr of rows in the screen
-EXTERN int Columns INIT(= DFLT_COLS);  // nr of columns in the screen
+EXTERN linenr_T Rows INIT(= DFLT_ROWS);     // nr of rows in the screen
+EXTERN colnr_T Columns INIT(= DFLT_COLS);  // nr of columns in the screen
```
Comment (zeertzjq) [inline src/nvim/globals.h:102]: ???

File: src/nvim/undo_defs.h [hunk 6]
```diff
@@ -56,7 +56,7 @@ struct u_header {
   u_entry_T *uh_entry;        // pointer to first entry
   u_entry_T *uh_getbot_entry;   // pointer to where ue_bot must be set
   pos_T uh_cursor;              // cursor position before saving
-  long uh_cursor_vcol;
+  colnr_T uh_cursor_vcol;
```
Comment (zeertzjq) [inline src/nvim/undo_defs.h:59]: This will break undo files.

**Review by zeertzjq — CHANGES_REQUESTED** [review]

File: src/nvim/globals.h (hunk 5 above)
Comment (dundargoc) [reply src/nvim/globals.h:102]: `linenr_T` is a type to describe number of rows. `Rows` is a variable describing the number of rows.

`colnr_T` is a type to describe number of columns. `Columns` is a variable describing the number of columns.

Comment (zeertzjq) [reply src/nvim/globals.h:102]: No. `linenr_T` and `colnr_T` and used to describe the number of text columns, not screen columns.

Comment (dundargoc) [reply src/nvim/globals.h:102]: If you say so. I don't think that's obvious. I've reverted it.

File: src/nvim/drawline.c [hunk 7]
```diff
@@ -2712,7 +2712,7 @@ int win_line(win_T *wp, linenr_T lnum, int startrow, int endrow, bool number_onl
           // 'search_hl' and the match list.
           get_search_match_hl(wp,
                               &screen_search_hl,
-                              (long)(ptr - line),  // NOLINT(google-readability-casting)
+                              (colnr_T)(ptr - line),  // NOLINT(google-readability-casting)
```
Comment (bfredl) [inline src/nvim/drawline.c:2715]: is this NOLINT also superfluous now?

File: src/nvim/statusline.c [hunk 8]
```diff
@@ -237,11 +237,11 @@ void stl_clear_click_defs(StlClickDefinition *const click_defs, const size_t cli
 }
 
 /// Allocate or resize the click definitions array if needed.
-StlClickDefinition *stl_alloc_click_defs(StlClickDefinition *cdp, long width, size_t *size)
+StlClickDefinition *stl_alloc_click_defs(StlClickDefinition *cdp, size_t width, size_t *size)
```
Comment (bfredl) [inline src/nvim/statusline.c:240]: variables which are a screen width like `width` can just be an `int`.

**Comment by bfredl** [comment]
Otherwise LGTM. nice cleanups!

File: src/nvim/cmdexpand.c [hunk 9]
```diff
@@ -444,7 +444,7 @@ static void redraw_wildmenu(expand_T *xp, int num_matches, char **matches, int m
   int row;
   char *buf;
   int len;
-  int clen;                     // length in screen cells
+  colnr_T clen;                     // length in screen cells
```
Comment (zeertzjq) [inline src/nvim/cmdexpand.c:447]: Change back to int

File: src/nvim/window.c [hunk 10]
```diff
@@ -5318,8 +5318,8 @@ static void frame_remove(frame_T *frp)
 
 void win_new_screensize(void)
 {
-  static long old_Rows = 0;
-  static long old_Columns = 0;
+  static linenr_T old_Rows = 0;
+  static colnr_T old_Columns = 0;
```
Comment (zeertzjq) [inline src/nvim/window.c:5322]:
```suggestion
  static int old_Rows = 0;
  static int old_Columns = 0;
```

**Comment by bfredl** [comment]
@dundargoc Please rebase and then we will merge this

**Comment by dundargoc** [comment]
@bfredl gtg?

**Comment by bfredl** [comment]
yep, after CI

**Pull request merged**